crates/rover-client/src/operations/subgraph/check_workflow/runner.rs

```diff
@@ -0,0 +1,149 @@
+use std::time::{Duration, Instant};
+
+use super::types::*;
+use crate::blocking::StudioClient;
+use crate::operations::subgraph::check_workflow::types::QueryResponseData;
+use crate::shared::{ChangeSeverity, CheckResponse, GraphRef, SchemaChange};
+use crate::RoverClientError;
+
+use apollo_federation_types::build::BuildError;
+
+use graphql_client::*;
+
+use self::subgraph_check_workflow_query::CheckWorkflowStatus;
+use self::subgraph_check_workflow_query::SubgraphCheckWorkflowQueryGraphCheckWorkflowTasksOn::{
+    CompositionCheckTask, OperationsCheckTask,
+};
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/operations/subgraph/check_workflow/check_workflow_query.graphql",
+    schema_path = ".schema/schema.graphql",
+    response_derives = "PartialEq, Debug, Serialize, Deserialize, Clone",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. subgraph_check_workflow_query
+pub(crate) struct SubgraphCheckWorkflowQuery;
+
+/// The main function to be used from this module.
+/// This function takes a proposed schema and validates it against a published
+/// schema.
+pub fn run(
+    input: CheckWorkflowInput,
+    subgraph: String,
+    client: &StudioClient,
+) -> Result<CheckResponse, RoverClientError> {
+    let graph_ref = input.graph_ref.clone();
+    let mut data;
+    let now = Instant::now();
+    // default timeout is 5 minutes
+    let timeout_seconds = option_env!("APOLLO_CHECKS_TIMEOUT_SECONDS")
```

Comment:
this variable should be read by `env.rs` and passed to this function as part of `CheckWorkflowInput`